src/plugins/controls/public/range_slider/components/range_slider_button.tsx

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { useCallback } from 'react';
+
+import { EuiFieldNumber, EuiFormControlLayoutDelimited } from '@elastic/eui';
+
+import './range_slider.scss';
+import { RangeValue } from '../../../common/range_slider/types';
+import { useRangeSlider } from '../embeddable/range_slider_embeddable';
+
+export const RangeSliderButton = ({
+  value,
+  onChange,
+  isPopoverOpen,
+  setIsPopoverOpen,
+}: {
+  value: RangeValue;
+  isPopoverOpen: boolean;
+  setIsPopoverOpen: (open: boolean) => void;
+  onChange: (newRange: RangeValue) => void;
+}) => {
+  const rangeSlider = useRangeSlider();
+
+  const min = rangeSlider.select((state) => state.componentState.min);
+  const max = rangeSlider.select((state) => state.componentState.max);
+  const isInvalid = rangeSlider.select((state) => state.componentState.isInvalid);
+
+  const id = rangeSlider.select((state) => state.explicitInput.id);
+
+  const isLoading = rangeSlider.select((state) => state.output.loading);
+
+  const onClick = useCallback(
+    (event) => {
+      // the popover should remain open if the click/focus target is one of the number inputs
+      if (isPopoverOpen && event.target instanceof HTMLInputElement) {
+        return;
+      }
+      setIsPopoverOpen(true);
+    },
+    [isPopoverOpen, setIsPopoverOpen]
+  );
```

Comment:
This is meant to replicate the behaviour of the `EuiDualRange` component:

![Jun-12-2023 09-45-15](https://github.com/elastic/kibana/assets/8698078/eca04d80-5c20-4fb3-aa90-759859a69d5c)

We will be able to clean this up once we are officially able to make the transition to the `EuiDualRange` component rather than building our own :)

Reply:
This keyboard interactivity downside is totally fine for the time being, especially since this is still better than it was, and we may not even release in this intermediate state. Nice work!